controlplane/kubeadm/internal/workload_cluster_conditions_test.go

```diff
@@ -1620,3 +1676,180 @@ func TestAggregateV1Beta2ConditionsFromMachinesToKCP(t *testing.T) {
+					fakeMachine("m2", withNodeRef("m2")),
+				),
+			},
+			members: []*etcd.Member{
+				{Name: "m1"},
+				{Name: "m2"},
+			},
+			nodes:                               nil,
+			expectMembersAndMachinesAreMatching: true,
+			expectKCPErrors:                     nil,
+		},
+		{
+			name: "true if there is a machine without a member but at least a machine is still provisioning",
+			controlPlane: &ControlPlane{
+				KCP: &controlplanev1.KubeadmControlPlane{},
+				Machines: collections.FromMachines(
+					fakeMachine("m1", withNodeRef("m1")),
+					fakeMachine("m2", withNodeRef("m2")),
+					fakeMachine("m3"), // m3 is still provisioning
+				),
+			},
+			members: []*etcd.Member{
+				{Name: "m1"},
+				{Name: "m2"},
+				// m3 is missing
+			},
+			nodes:                               nil,
+			expectMembersAndMachinesAreMatching: true,
+			expectKCPErrors:                     nil,
+		},
+		{
+			name: "true if there is a machine without a member but node on this machine does not exist yet",
+			controlPlane: &ControlPlane{
+				KCP: &controlplanev1.KubeadmControlPlane{},
+				Machines: collections.FromMachines(
+					fakeMachine("m1", withNodeRef("m1")),
+					fakeMachine("m2", withNodeRef("m2")),
+					fakeMachine("m3", withNodeRef("m3")),
+				),
+			},
+			members: []*etcd.Member{
+				{Name: "m1"},
+				{Name: "m2"},
+				// m3 is missing
+			},
+			nodes: &corev1.NodeList{Items: []corev1.Node{
+				// m3 is missing
+			}},
+			expectMembersAndMachinesAreMatching: true,
+			expectKCPErrors:                     nil,
+		},
+		{
+			name: "true if there is a machine without a member but node on this machine has been just created",
+			controlPlane: &ControlPlane{
+				KCP: &controlplanev1.KubeadmControlPlane{},
+				Machines: collections.FromMachines(
+					fakeMachine("m1", withNodeRef("m1")),
+					fakeMachine("m2", withNodeRef("m2")),
+					fakeMachine("m3", withNodeRef("m3")),
+				),
+			},
+			members: []*etcd.Member{
+				{Name: "m1"},
+				{Name: "m2"},
+				// m3 is missing
+			},
+			nodes: &corev1.NodeList{Items: []corev1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "m3", CreationTimestamp: metav1.Time{Time: time.Now().Add(-110 * time.Second)}}}, // m3 is just provisioned
+			}},
+			expectMembersAndMachinesAreMatching: true,
+			expectKCPErrors:                     nil,
+		},
+		{
+			name: "false if there is a machine without a member and node on this machine is old",
+			controlPlane: &ControlPlane{
+				KCP: &controlplanev1.KubeadmControlPlane{},
+				Machines: collections.FromMachines(
+					fakeMachine("m1", withNodeRef("m1")),
+					fakeMachine("m2", withNodeRef("m2")),
+					fakeMachine("m3", withNodeRef("m3")),
+				),
+			},
+			members: []*etcd.Member{
+				{Name: "m1"},
+				{Name: "m2"},
+				// m3 is missing
+			},
+			nodes: &corev1.NodeList{Items: []corev1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "m3", CreationTimestamp: metav1.Time{Time: time.Now().Add(10 * time.Minute)}}}, // m3 is old
```

Comment:
```suggestion
				{ObjectMeta: metav1.ObjectMeta{Name: "m3", CreationTimestamp: metav1.Time{Time: time.Now().Add(-10 * time.Minute)}}}, // m3 is old
```

expectMembersAndMachinesAreMatching should then be false